app/questionnaire/path_finder.py

```diff
@@ -59,19 +59,32 @@ def build_path(self):
                 group_blocks = self._build_blocks_for_group(group, instance_idx)
                 blocks += group_blocks
 
-            if group['id'] in first_groups:
-                this_location = Location(group['id'], 0, group['blocks'][0]['id'])
+            first_group_instance_index = self._get_first_group_instance_index_in_block(group, blocks)
+            if first_group_instance_index is not None:
+                if group['id'] in first_groups:
+                    this_location = Location(group['id'], first_group_instance_index, group['blocks'][0]['id'])
 
-            if blocks:
-                path, block_index = self._build_path_within_group(blocks, block_index, this_location, path)
+                if blocks:
+                    path, block_index = self._build_path_within_group(blocks, block_index, this_location, path)
 
         return path
 
+    @staticmethod
+    def _get_first_group_instance_index_in_block(group, blocks):
+        """returns the first group instance of a block that does not need to be skipped
```

Comment:
The method code doesn't match the docstring. This method simply returns the group_instance index of the first block belonging to the group. Perhaps this might be better as an inline comment on the calling code to better explain how `first_group_instance_index` is being used?

Reply:
Good catch - thanks